tests/unit/build_providers/lxd/test_lxd.py

```diff
@@ -159,23 +157,22 @@ def test_create(self):
                     "alias": "16.04",
                 },
                 "environment.SNAPCRAFT_HAS_TTY": "False",
-                "user.user-data": "fake-cloud",
             },
             wait=True,
         )
 
         container = self.fake_pylxd_client.containers.get(self.instance_name)
         container.start_mock.assert_called_once_with(wait=True)
         self.assertThat(container.save_mock.call_count, Equals(2))
-        self.assertThat(container.sync_mock.call_count, Equals(3))
-        self.assertThat(self.check_call_mock.call_count, Equals(2))
+        self.assertThat(container.sync_mock.call_count, Equals(11))
+        self.assertThat(self.check_call_mock.call_count, Equals(8))
```

Comment:
I've noticed this pattern across the tests.  I've been trying to use the approach of checking `mock_calls` instead of `assert_has_calls` + matching a tally for `call_count`. e.g.:

```
self.assertThat(self.check_call_mock.mock_calls, Equals(<call-list>))
```

Not that you have to change it here, I'm mostly curious if you are choosing that pattern on purpose, or just didn't realize about `mock_calls`?

Reply:
assert_has_calls does not require you to ensure that all calls are there, very useful at times.